pkg/funky/router.go

```diff
@@ -76,14 +76,15 @@ func (r *DefaultRouter) Delegate(input Message) (*Message, error) {
 	if err != nil {
 		switch err.(type) {
 		case TimeoutError:
-			server.Terminate()
-			server, serverErr := r.serverFactory.CreateServer(server.GetPort())
-			if serverErr != nil {
-				Healthy = false
+			terminateErr := server.Terminate()
+			newServer, serverErr := r.serverFactory.CreateServer(server.GetPort())
+			if serverErr != nil || terminateErr != nil {
+				Healthy <- false
```

Comment:
Sending on a closed channel results in a panic. It would be rare, but if funky is processing multiple invocations they could trigger this multiple times and kill funky.